### DISCUSSION TITLE
Can you call REST API from mkdocs?

### DISCUSSION THREAD

**Discussion by dharmeshpurohit** [mkdocs/mkdocs#2881] [Q&A]
Is there a way I can call REST API from mkdocs? Any plugins that I can use for this perpose?

**Comment by pawamoy** [comment]
JavaScript ? Could you expand on your use-case :smile:?

**Reply by dharmeshpurohit** [reply]
Say for example I have markdown page status.md. On that page I want to call status api and display status of system availability based on the response I get from API.

**Reply by dharmeshpurohit** [reply]
It doesn't have to be `.md` file. I can create Javascript/HTML page if there is a way to call api

**Reply by pawamoy** [reply]
You can definitely add Javascript code to your Markdown file, just like in a HTML file.

**Reply by dharmeshpurohit** [reply]
I would really appreciate if you can point me to examples.

**Reply by dharmeshpurohit** [reply]
Using following I can add but is there way I can use Javascript or NodeJS code from markdown?
```js
 ```js
   const myvar = "hello"
   module.exports.response = response = ()=>{mycode here}
    ```

**Comment by flefevre** [comment]
I am interested by the same feature. My idea is to be able to display in my mydocs web site
The status of git project
List of tickets...

For that I was thinking to retrieve those elements by calling rest API of GitHub or other tools such as Tuleap.

I would love to have a basic example.

**Comment by pawamoy** [comment]
Build time solution with [Markdown-Exec](https://pawamoy.github.io/markdown-exec/):

````md
List of issues for this repository:

```bash exec="1" result="ansi"
gh issue list
```
````

Instead of a Bash executed code block, you could use Python to query the API too:

````md
List of issues for this repository:

```python exec="1"
import httpx

with httpx.Client() as client:
    response = client.get("https://api.github.com/...")
    data = response.json()
    # etc.
print(data)
```
````
